File: kinotic-domain/src/main/java/org/kinotic/os/api/model/iam/AuthType.java
package org.kinotic.os.api.model.iam;

/**
 * Authentication method for an {@link IamUser}.
 * LOCAL uses email/password with bcrypt-hashed credentials.
 * OIDC uses federated identity via an external provider (Google, Microsoft, etc.).
 */
public enum AuthType {
    LOCAL,
    OIDC
}

File: kinotic-js/workspace/packages/os-api/src/api/model/security/AuthType.ts
/**
 * Authentication method for an {@link ParticipantIdentity}.
 */
export enum AuthType {
    /** Email/password with bcrypt-hashed credentials. */
    LOCAL = 'LOCAL',
    /** Federated identity via an external OIDC provider (Google, Microsoft, etc.). */
    OIDC = 'OIDC',
    /**
     * Only Kinotic-issued tokens minted when a USER authorizes the delegate; a DELEGATED
     * identity has no login credential of its own.
     */
    DELEGATED = 'DELEGATED',
    /**
     * A provisioned client id and secret presented as connection credentials; the
     * authentication method of every MACHINE identity.
     */
    CLIENT_CREDENTIALS = 'CLIENT_CREDENTIALS'
}
